### PR TITLE
Add 'hidden' optional property to DeclarativeTableColumn

### PR DESCRIPTION
**Pull request by rchlkm**
This PR adds support to dynamically hide DeclarativeTableColumn in the DeclarativeTableComponent. 

The snippet below will hide column index 5.
```
table.columns[5].hidden = true;
```

This will be used for the SQL Migration extension to show/hide columns based on user input. This will avoid the need to recreate the table each time the user input changes.

Show All Column
image

Hide Column Index=5
image

### CODE REVIEW

**Comment by Charles-Gagnon** [comment]
What "user input" causes the table to change? The screenshots you show don't really indicate what it is that's triggering the columns to be hidden.

**Comment by rchlkm** [comment]
> What "user input" causes the table to change? The screenshots you show don't really indicate what it is that's triggering the columns to be hidden.

@Charles-Gagnon In the context of the SQL Migration extension, on Step 3 of the wizard, user will select the migration mode (Online or Offline). In the case of an Online migration mode, the File Name column needs to be removed. 
image

When the Migration wizard is first opened, this DeclarativeTableComponent is created with all the columns. However, once the user selects "Online migration", the table should hide the File Name column.

**Comment by Charles-Gagnon** [comment]
It seems like having a `hidden` property on the column that could be modified would be a cleaner way of handling this - having to have callers deal with indexes isn't as intuitive. 

At some point I think it's likely that we'll want to add a column filtering option to these tables, and when/if that happens having a property on the columns would allow the extensions to also check whether a column has been hidden by the user without relying on events to inform them.

**Comment by rchlkm** [comment]
I changed the logic based on your feedback by adding the 'hidden' property on the columns

File: src/sql/workbench/browser/modelComponents/declarativeTable.component.ts [hunk 1]
```diff
@@ -466,4 +466,8 @@ export default class DeclarativeTableComponent extends ContainerBase<any, azdata
 			this.setPropertyFromUI<number>((properties, value) => { properties.selectedRow = value; }, row);
 		}
 	}
+
+	public showColumn(column: azdata.DeclarativeTableColumn): boolean {
+		return column.hidden === undefined || !column.hidden;
```
Comment (Charles-Gagnon) [inline src/sql/workbench/browser/modelComponents/declarativeTable.component.ts:471]: You can just do `!column.hidden` - that'll handle both of these cases.

**Review by Charles-Gagnon — APPROVED** [review]

**Review by alanrenmsft — APPROVED** [review]

**Review by brian-harris — APPROVED** [review]

**Pull request merged**